compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8828,34 +8839,61 @@ private STNode parseKeyTypeConstraint(STNode keyKeywordToken) {
     /**
      * Parse function type descriptor.
      * <p>
-     * <code>function-type-descriptor := function function-signature</code>
+     * <code>function-type-descriptor := [isolated] function function-signature</code>
      *
      * @return Function type descriptor node
      */
     private STNode parseFunctionTypeDesc() {
         startContext(ParserRuleContext.FUNC_TYPE_DESC);
+        List<STNode> qualifiers = new ArrayList<>();
+        STToken nextToken = peek();
+        switch (nextToken.kind) {
+            case ISOLATED_KEYWORD:
+                qualifiers.add(consume());
+                break;
+            case FUNCTION_KEYWORD:
+                break;
+            default:
+                recover(peek(), ParserRuleContext.FUNC_TYPE_DESC_START);
+                return parseFunctionTypeDesc();
+        }
```

Comment:
Seems this block below is repeated in 3 places. Shall we extract it to a separate method?

Reply:
moved to parseFunctionQualifiers method